mobile_packet_verifier/src/verifier.rs

```diff
@@ -0,0 +1,44 @@
+use chrono::{DateTime, Utc};
+use file_store::mobile_session::DataTransferSessionIngestReport;
+use futures::{Stream, StreamExt};
+use sqlx::{Postgres, Transaction};
+
+#[derive(thiserror::Error, Debug)]
+pub enum VerificationError {
+    #[error("file store error: {0}")]
+    FileStoreError(#[from] file_store::Error),
+    #[error("sqlx error: {0}")]
+    SqlxError(#[from] sqlx::Error),
+    #[error("reports stream dropped")]
+    ReportsStreamDropped,
+}
+
+pub async fn verify(
+    conn: &mut Transaction<'_, Postgres>,
+    curr_file_ts: DateTime<Utc>,
+    reports: impl Stream<Item = DataTransferSessionIngestReport>,
+) -> Result<(), VerificationError> {
+    tokio::pin!(reports);
+
+    while let Some(DataTransferSessionIngestReport { report, .. }) = reports.next().await {
+        sqlx::query(
+            r#"
+            INSERT INTO data_transfer_sessions (pub_key, payer, uploaded_bytes, downloaded_bytes, first_timestamp, last_timestamp)
+            VALUES ($1, $2, $3, $4, $5, $5)
+            ON CONFLICT (pub_key, payer) DO UPDATE SET
```

Comment:
is the intention that within a given burn period any new data transfer for a given `(pub_key, payer)` will get appended to the existing up/down totals and the `last_timestamp` updated to the latest and that _after_ a given burn period the records in the db table are wiped? what happens if there are more than one "session" of data transfer within a given period for a single `(pub_key, payer)` combo?

Reply:
I'm not sure I understand the question, for the given period all of the data_transfer sessions are accumulated. Perhaps I need to change the `last_timestamp` to the max of the current last timestamp and the new one?

Reply:
you answered my question (there's only one "data transfer session" per burn period and the table is wiped after each period. i think catching the `MAX` of the timestamps is a good catch though in case the files maybe got received/processed out of order?

Reply:
It shouldn't really matter because it's the file timestamp and no the session timestamp